image_explorer/image_explorer.py

```diff
@@ -33,6 +33,8 @@
 
 from django.conf import settings
 from lxml import etree, html
+# https://github.com/openedx/edx-platform/blob/open-release/sumac.2/requirements/edx/base.txt#L718
+etree.__version__ = "4.9.4"
```

Comment:
This seems a bit too hacky to me. Checking a local install of lxml I see that ``etree.__version__`` already has the correct value, so what is this patch doing?

Currently, we're installing an unconstrained version of parsel, perhaps we need to pin it?

This is going to cause issue elsewhere since we're monkey patching the version and that means the effect could be somewhere far from the cause. 

Also why are the tests working without this patch? Could it be because the block is being tested with lxml>=5 I think we need to look for a different solution.

Reply:
@xitij2000 did you try getting etree.__version__ inside the cms/lms shell?  We already patch lxml inside the edx-platform so that we don't expose certain functionality.

Reply:
That is unfortunate. 

I think the issue here is that we are not pinning the version of lxml or parsel to something that's compatible with edx-platform. 

I think if you pin parsel to v1.6 that will also resolve this issue without needing to update edx-platform. 

I think parsel and lxml should be pinned in this repo anyway to a version we've tested.

Reply:
Will pinning solve the problem? I mean we would have to update someday, right?

Reply:
LXML is pinned to <5 in the platform so we shouldn't be using v5 with the block anyway. We certainly will have to update someday, but at that point the monkeypatches in the platform that are causing this issue might also be gone. 

We know the older version works, we don't know what else the monkeypatches are doing that might break other things in the block that we don't know about yet.

Reply:
Alright I have pinned the dependencies here. I think https://github.com/openedx/edx-platform/pull/36627 still makes sense but I don't know if we should spend time on that now.